libbinlogevents/src/statement_events.cpp

```diff
@@ -532,8 +532,7 @@ User_var_event(const char* buf, unsigned int event_len,
                          (description_event->footer()->checksum_alg ==
                           BINLOG_CHECKSUM_ALG_OFF));
   size_t data_written= (header()->data_written- checksum_verify);
-  BAPI_ASSERT(((bytes_read == data_written) ? 0 : BINLOG_CHECKSUM_LEN)||
-              ((bytes_read == data_written - 1) ? 0 : BINLOG_CHECKSUM_LEN));
+  BAPI_ASSERT(bytes_read == data_written || bytes_read == data_written);
```

Comment:
Ops, I wanted to write data_written -1 for the second part.

Based on the code (and comment) above, I think this assert wants to ensure that we had read the same amount as we have written, or maybe one byte more in some cases.

Reply:
But now that I checked it again, maybe BINLOG_CHECKSUM_LEN should be included in this calculation, I'm not really sure :( I will try to test it.

Unfortunately, the original condition always evaluates to true... (or rather to BINLOG_CHECKSUM_LEN)

Reply:
I ran the tests with my version, and it results in more failures - I could keep it in the state as it's in 8.0 (which compiles, but always evaluates to true), or maybe  I could simply remove it?

Reply:
Looking at the 8.0 version - ```BAPI_ASSERT(((bytes_read == data_written) ? false : true) ||
((bytes_read == data_written - 1) ? false : true));``` - it is not obvious to me why would it always evaluate to true?

Reply:
If (bytes_read == data_written - 1) is true, then the first check has to be false, and since the ? : expression negates it, it results in a false.
If (bytes_read == data_written) is true, the same happens in the other direction: the other check will be false, and the expression will negate it.

But I found out what this expression wants to check based on the 5.6 branch, I'm testing it currently.